src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets

```diff
@@ -108,11 +108,10 @@ Copyright (c) .NET Foundation. All rights reserved.
     <DefaultNetCorePatchVersion Condition="'$(_TargetFrameworkVersionWithoutV)' == '1.0'">1.0.5</DefaultNetCorePatchVersion>
     <DefaultNetCorePatchVersion Condition="'$(_TargetFrameworkVersionWithoutV)' == '1.1'">1.1.2</DefaultNetCorePatchVersion>
 
-    <DefaultNetCorePatchVersion Condition="'$(_TargetFrameworkVersionWithoutV)' == '2.1'">2.1.0</DefaultNetCorePatchVersion>
-
     <!-- If targeting the same pre-release that is bundled with the .NET Core SDK, use the bundled package version
-         provided by Microsoft.NETCoreSdk.BundledVersions.props -->
-    <DefaultNetCorePatchVersion Condition="'$(DefaultNetCorePatchVersion)' == '' and '$(_TargetFrameworkVersionWithoutV)' == '$(BundledNETCoreAppTargetFrameworkVersion)'">$(BundledNETCoreAppPackageVersion)</DefaultNetCorePatchVersion>
+         provided by Microsoft.NETCoreSdk.BundledVersions.props. Note there is a dot appended to _TargetFrameworkVersionWithoutV for comparison to fix major and minor -->
+    <_TargetFrameworkVersionWithoutVMatchesMajorAndMinor Condition="$(BundledNETCoreAppPackageVersion.StartsWith('$(_TargetFrameworkVersionWithoutV).'))">true</_TargetFrameworkVersionWithoutVMatchesMajorAndMinor>
```

Comment:
Use `'$(_TargetFrameworkVersionWithoutV)' == '$(BundledNETCoreAppTargetFrameworkVersion)'` instead of the clause that checks whether the bundled version starts with `$(TargetFrameworkVersion).`.

Reply:
Fixed